packages/core/src/common/utils.ts

```diff
@@ -33,6 +35,10 @@ export function getDisplayName(ComponentClass: React.ComponentClass | INamed) {
     return (ComponentClass as React.ComponentClass).displayName || (ComponentClass as INamed).name || "Unknown";
 }
 
+export function isElementType(element: JSX.Element, ComponentClass: React.ComponentClass) {
```

Comment:
would be awesome to do something like this:
```
isElementType<P = {}>(element: any, ComponentClass: React.ComponentClass<P>): element is React.ReactElement<P>
```

then you won't need to cast when calling this function (but you will need to check `!= null` and cast inside this fn)